library/alloc/src/vec/mod.rs

```diff
@@ -478,7 +478,7 @@ impl<T> Vec<T> {
     #[stable(feature = "rust1", since = "1.0.0")]
     #[must_use]
     pub fn with_capacity(capacity: usize) -> Self {
-        Self::with_capacity_in(capacity, Global)
+        Vec { buf: RawVec::with_capacity_in(capacity, Global), len: 0 }
```

Comment:
Why did you change this?

Reply:
It removes another generic function to inline on a hot path. This is likely where majority of the improvement comes from.

Reply:
I'm not entirely convinced of this; the MIR inliner should be really good at these call-forwarding things. But I can experiment on that after this has merged.

Reply:
The MIR inliner should be really good at simple bodies like this, so I'm not convinced of that. Could you, for the sake of experimentation, undo this change?

Reply:
@joboet Are you satisfied with the result? I don't know if the small difference is a measurement noise or not.